### ISSUE TITLE
Bundle / development package import

### ISSUE THREAD

**Issue by iakremnev** [bentoml/BentoML#1621]
**Describe the bug**

Required modules from the local development package are not imported to the service bundle when packing a service with artifacts.

**To Reproduce**

1. Clone https://github.com/iakremnev/bentoml-issue-example
2. Make virtual environment and install local package in development mode: 
    ```
    python -m venv venv
    source venv/bin/activate
    pip install -e .
    ```
3. Pack the service:
    ```
    python export_service.py
    ```
4. Try and run the service:
    ```
    bentoml serve Service:latest
    ```

As a result, the service can't import `mypackage.model` since `model.py` is not bundled.

**Expected behavior**

Code for model class definition and any preprocessing is bundled with the service.

**Environment:**
 - OS: Arch Linux
 - Python 3.9.2
 - BentoML 0.12.1

**Additional context**

**Comment by iakremnev** [comment]
I've seen this and related issues mentioned quite a few times #1410, #608, [in slack ](https://bentoml.slack.com/archives/CKRANBHPH/p1604421962371200). The suggested workaround is to import all the required modules in the module with BentoService class definition.
So it's rather a feature suggestion, than a bug report: **trace required modules for the artifact that is packed**.

**Comment by yubozhao** [comment]
@iakremnev Thank you for the reproducible repo!  We are working on new user facing APIs that will make it a lot easier for users to include files

**Comment by parano** [comment]
This issues has been resolved in the upcoming BentoML 1.0 release.

In BentoML 1.0, a project root must be specifically defined by placing a bentofile.yaml file in the directory. This file specifies all the Bento build configs, including what are the files to include in the final Bento built. The project root also should be seen as the CWD in your service's python environment, as well as part of the import path in sys.path. A more detailed explanation can be found here: https://github.com/bentoml/BentoML/tree/main/bentoml/_internal/bento
